frontend/src/pages/ProjectSettings/GitHubSettingsModal/GitHubSettingsModal.tsx

```diff
@@ -0,0 +1,190 @@
+
+import * as styles from './GitHubSettingsModal.css'
+
+type Props = {
+	service: Service
+	githubRepos: GitHubRepo[]
+	handleSave: (service: Service, repo: string | null) => void
+	githubIntegrated: boolean
+	closeModal: () => void
+}
+
+type FormValues = {
+	githubRepo: string | null
+}
+
+export const GitHubSettingsModal = ({
+	service,
+	githubRepos,
+	handleSave,
+	githubIntegrated,
+	closeModal,
+}: Props) => {
+	const handleSubmit = (formValues: FormValues) => {
+		handleSave(service, formValues.githubRepo)
+		closeModal()
+	}
+
+	if (!service) {
+		return null
+	}
+
+	return (
+		<Modal
+			onCancel={closeModal}
+			visible={!!service}
+			title={`GitHub settings for ${service.name} service`}
+			width="600px"
+		>
+			<ModalBody>
+				{!githubIntegrated ? (
+					<Stack>
+						<Text size="small" weight="medium">
+							It looks like you have not enabled GitHub to
+							integrate with Highlight yet. Visit{' '}
+							<Link to={`/${service.projectID}/integrations`}>
+								Integrations
+							</Link>{' '}
+							to start creating GitHub issues from comments,
+							enhancing your error stacktraces with more context,
+							and more!
+						</Text>
+						<Box display="flex">
+							<LinkButton
+								to={`/${service.projectID}/integrations`}
+								trackingId="github-settings-integrations-page-link"
+							>
+								Integrate GitHub
+							</LinkButton>
+						</Box>
+					</Stack>
+				) : (
+					<GithubSettingsForm
+						service={service}
+						githubRepos={githubRepos}
+						handleSubmit={handleSubmit}
+						handleCancel={closeModal}
+					/>
+				)}
+			</ModalBody>
+		</Modal>
+	)
+}
+
+type GithubSettingsFormProps = {
+	service: Service
+	githubRepos: GitHubRepo[]
+	handleSubmit: (formValues: FormValues) => void
+	handleCancel: () => void
+}
+
+const GithubSettingsForm = ({
+	service,
+	githubRepos,
+	handleSubmit,
+	handleCancel,
+}: GithubSettingsFormProps) => {
+	const githubOptions = useMemo(
+		() =>
+			githubRepos.map((repo: GitHubRepo) => ({
+				id: repo.key,
+				label: repo.name.split('/').pop(),
+				value: repo.repo_id.replace(
+					'https://api.github.com/repos/',
+					'',
+				),
+			})),
+		[githubRepos],
+	)
+
+	const form = Form.useFormState<FormValues>({
+		defaultValues: { githubRepo: service.githubRepoPath || null },
+	})
+
+	return (
+		<Form state={form} onSubmit={() => handleSubmit(form.values)}>
+			<Stack>
+				<Text size="small">
+					Connect <i>{service.name}</i> to a GitHub repository to
+					enable enhancements when debugging errors.
+				</Text>
+				<Box display="flex" alignItems="center" gap="16">
+					<Select
+						aria-label="GitHub Repository"
+						className={styles.repoSelect}
+						placeholder="Search repos..."
+						onSelect={(repo: string) =>
+							form.setValue(form.names.githubRepo, repo)
+						}
+						value={form.values.githubRepo?.split('/').pop()}
+						options={githubOptions}
+						notFoundContent={<span>No repos found</span>}
```

Comment:
I guess this can happen if they don't grant us access to any of their repos during the auth flow. Could consider calling this out somewhere.

Reply:
Yep, its also the case if they type in some nonsense or a repo that doesn't exist

<img width="616" alt="Screenshot 2023-08-11 at 5 50 49 PM" src="https://github.com/highlight/highlight/assets/17744174/cd7679fa-e10c-4e06-9972-193387d357e1">